app/controllers/application_controller/performance.rb

```diff
@@ -183,274 +183,326 @@ def perf_menu_click
-          page << 'ManageIQ.charts.chartData = ' + {
-            "candu"  => @chart_data,
-            "parent" => @parent_chart_data
-          }.to_json + ';'
-        else
-          page << 'ManageIQ.charts.chartData = ' + {
-            "candu" => @chart_data
-          }.to_json + ';'
-        end
-        page.replace("perf_options_div", :partial => "layouts/perf_options")
-        page.replace("candu_charts_div", :partial => "layouts/perf_charts", :locals => {:chart_data => @chart_data, :chart_set => "candu"})
-        page << js_build_calendar(@perf_options.to_calendar)
-        page << Charting.js_load_statement
-        page << 'miqSparkle(false);'
-      end
-      return
-
-    elsif cmd == "Chart" && model == "Selected"                             # Create daily/hourly chart for selected CI
-      return unless @record = perf_menu_record_valid(data_row["resource_type"], data_row["resource_id"], data_row["resource_name"])
-      # Set the perf options in the selected controller's sandbox
-      cont = data_row["resource_type"].underscore.downcase.to_sym
-      session[:sandboxes][cont] ||= {}
-      session[:sandboxes][cont][:perf_options] ||= Options.new
-
-      # Copy general items from the current perf_options
-      session[:sandboxes][cont][:perf_options][:index] = @perf_options[:index]
-      session[:sandboxes][cont][:perf_options][:tz] = @perf_options[:tz]
-      session[:sandboxes][cont][:perf_options][:time_profile] = @perf_options[:time_profile]
-      session[:sandboxes][cont][:perf_options][:time_profile_days] = @perf_options[:time_profile_days]
-      session[:sandboxes][cont][:perf_options][:time_profile_tz] = @perf_options[:time_profile_tz]
-
-      # Set new perf options based on what was selected
-      session[:sandboxes][cont][:perf_options][:model] = data_row["resource_type"]
-      session[:sandboxes][cont][:perf_options][:typ] = typ
-      session[:sandboxes][cont][:perf_options][:daily_date] = @perf_options[:daily_date] if typ == "Daily"
-      session[:sandboxes][cont][:perf_options][:days] = @perf_options[:days] if typ == "Daily"
-      session[:sandboxes][cont][:perf_options][:hourly_date] = [ts.month, ts.day, ts.year].join("/") if typ == "Hourly"
-
-      if data_row["resource_type"] == "VmOrTemplate"
-        prefix = TreeBuilder.get_prefix_for_model(@record.class.base_model)
-        tree_node_id = "#{prefix}-#{@record.id}"  # Build the tree node id
-        session[:exp_parms] = {:display => "performance", :refresh => "n", :id => tree_node_id}
-        javascript_redirect :controller => data_row["resource_type"].underscore.downcase.singularize,
-                            :action     => "explorer"
+    render :update do |page|
+      page << javascript_prologue
+      if @parent_chart_data
+        page << 'ManageIQ.charts.chartData = ' + {
+          "candu"  => @chart_data,
+          "parent" => @parent_chart_data
+        }.to_json + ';'
+      elsif @parent_chart_data
```

Comment:
should this be `elsif @compare_vm_chart_data` instead of never true?

(given https://github.com/ManageIQ/manageiq-ui-classic/pull/652/files#diff-d326989f8d12f33be175da2240bca798R34)